app/frontend/src/components/EditUserLocationMap.tsx

```diff
@@ -77,9 +77,15 @@ export default function EditUserLocationMap({
 
     map.current!.getCanvas().style.cursor = "grab";
 
-    const moveEvent = (e: MapMouseEvent | MapTouchEvent) => onCircleMove(e);
-    map.current!.on("mousemove", moveEvent);
-    map.current!.once("mouseup", (e) => onCircleUp(e, moveEvent));
+    if (e.type === "touchstart") {
```

Comment:
Couldn't we attach the event handlers to all of these events without this if-else branching? Or will that cause the mouse event to fire twice?

Reply:
The mouse move event gets removed in the mouse up event, so I'm worried about adding a bunch of touch move events that never get removed

Reply:
There's a `map.current!.off("touchmove", ...)` straight after the same for `mousemove`. Would that not remove it on every call of `onCircleUp`?

Reply:
On the topic of removing handlers... shouldn't we also be removing those assigned in `initializeMap`? Otherwise we could get some bad memory leaks each time a map is created...

Reply:
I just tried this. I occasionally got weird behaviour without the if branching, I guess it was double-firing.

I tried adding a Destructor return function, but it's no good and also not needed because the `postMapInitialize` `useEffect` returns immediately if the map already exists

Reply:
That sounds like that early `useEffect` exit thing is stopping us from cleaning up the map if it unmounts.

In their (only) React tutorial it looks like there's a `map.remove()` function that'd clean up everything the map holds onto when called. Can we call that in this component with a `useEffect` like:

```typescript
useEffect(() => {
    return () => {
      map.current!.remove();
    };
}, []);
```
`map` is a ref so the object identity is stable and so the lint rule shouldn't complain about missing it out.

Reply:
Done